lib/spack/spack/test/compilers/detection.py

```diff
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 """Test detection of compiler version"""
 import os
+import sys
```

Comment:
In general, I think these tests should not use `check_package_detection`, but we should instead:
1. Write `detection_tests.yaml` for each compiler
2. Add the features that are currently missing to `spack audit externals` (which are them?)

Here we are importing a lot of `builtin` packages, which couples the tests with the `builtin` repository, and is not scalable. In my opinion one objective would be to have packages that can be used as a compiler and carry their own test wherever repository they are located in.

Reply:
Done